### PR TITLE
Update Providers Reports index

### PR DESCRIPTION
**Pull request by inulty-dfe**
# Context

Update the reports section in the Provider Interface to show link to the new Recruitment Performance Report and remove the link to the old mid-cycle report

## Changes proposed in this pull request
The page has multiple potential states. The context is always for a particular provider user.

1. Provider User has one provider they can access reports for
2. Provider User has multiple providers they can access reports for
3. Each provider may or may not have a mid-cycle report generated
4. Each provider may or may not have a recruitment performance report generated
5. When the FeatureFlag is enabled, we show any recruitment performance reports but not the mid-cycle reports
6. When the FeatureFlag is disabled, we show any mid-cycle reports but not the recruitment performance reports 

## One provider for the user
|Performance Report **active**|Performance Report **inactive**|
|---|---|
|image|image|

## Multiple providers for the user
|Performance Report **active**|Performance Report **inactive**|
|---|---|
|image|image|

In order to calculate the date up to which the report counts candidates, I've added `CycleTimetable.cycle_week_date_range`. We can get the first and last day of the cycle week from this method as a bonus.

## Guidance to review

Providers 8 and 1 have recruitment performance reports generated for them
https://apply-review-9359.test.teacherservices.cloud/support/providers/8/users

## Link to Trello card

[Trello Ticket](https://trello.com/c/kI7LeG3D/1634-update-provider-reports-view)

## Things to check

- [ ] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] If this code adds a column to the DB, decide whether it needs to be in analytics yml file or analytics blocklist
- [ ] API release notes have been updated if necessary
- [ ] If it adds a significant user-facing change, is it documented in the [CHANGELOG](CHANGELOG.md)?
- [ ] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

File: app/models/publications/provider_recruitment_performance_report.rb [hunk 1]
```diff
@@ -2,5 +2,9 @@ module Publications
   class ProviderRecruitmentPerformanceReport < ApplicationRecord
     belongs_to :provider
     validates :cycle_week, :publication_date, presence: true
+
+    def reporting_date
```
Comment (elceebee) [inline app/models/publications/provider_recruitment_performance_report.rb:6]: I think the name of this one is confusing. I would think 'reporting date' would be the date the report was created, which is the publications date. Maybe `reporting_end_date`. `report_range_end_date`

File: app/views/provider_interface/reports/index.html.erb [hunk 2]
```diff
@@ -5,46 +5,69 @@
...
-        <li>
-          <%= govuk_link_to t('page_titles.provider.withdrawal_report'), provider_interface_reports_provider_withdrawal_report_path(provider_id: @providers.first) %>
-        </li>
-        <% if mid_cycle_report_present_for?(@providers.first) %>
-          <%= govuk_link_to mid_cycle_report_label_for(@providers.first), provider_interface_reports_provider_mid_cycle_report_path(provider_id: @providers.first) %>
+
+    <% if FeatureFlag.active?(:recruitment_performance_report) %>
+      <% if @performance_reports %>
+        <h2 class="govuk-heading-m">
+          Weekly recruitment performance report
```
Comment (elceebee) [inline app/views/provider_interface/reports/index.html.erb:12]: I think we should be consistent with the locales and put this in a locale file.

**Review by RebeccaCryan — APPROVED** [review]

**Review by elceebee — APPROVED** [review]
I've tested this locally and it all looks good to me. I have a couple of minor suggestions, but they are not blockers.

**Review by elceebee — APPROVED** [review]
🚀

**Pull request merged**